### PR TITLE
Further work on the dashboard

### PR DESCRIPTION
**Pull request by leekelleher**
Hey @mzajkowski, I did some more work on the dashboard part...

- Added API controller to get the themes (from disk)
- Parsing the CSS files to extract the theme name/description (a la WordPress style)
- Included button to unload the current theme
- Added "Invert" theme, as an example - it can be removed later

### CODE REVIEW

File: src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
+﻿using System.IO;
+using Umbraco.Core;
+
+namespace Our.Umbraco.BackofficeThemes.Models
+{
+    internal class ThemeInfo
+    {
+        public string alias { get; set; }
```
Comment (mzajkowski) [inline src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs]: @leekelleher why are you keeping those variables names lowercased?

File: src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            for (int i = 0; i < lines.Length; i++)
+            {
+                var line = lines[i];
+
+                // the first line must been an opening comment
+                if (i == 0 && line.StartsWith("/*") == false)
+                    return null;
+
+                // as soon as we reach a closing comment, exit
+                if (line.StartsWith("*/"))
```
Comment (mzajkowski) [inline src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs:48]: Interesting approach. We've been using .manifest JSON file for something like it in the past, but here we can definitely simplify it to this (at least for now).

**Review by mzajkowski — CHANGES_REQUESTED** [review]
Yet another good additions! I would only change the variables names to not start from lowercased letters and we can proceed further with it.

File: src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs (hunk 1 above)
Comment (leekelleher) [reply src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs]: I was just being lazy 😉
I wanted them lowercase on the client side, but didn't add the JsonProperty attributes.

I'll fix it up (tomorrow).

File: src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs (hunk 2 above)
Comment (leekelleher) [reply src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs:48]: I'm not fixed on this, happy with alternative approaches.
I was purely copying the WordPress way of theme meta data.

File: src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs (hunk 1 above)
Comment (mzajkowski) [reply src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs]: Ha good excuse! :) Cool, thanks in advance.

File: src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs (hunk 2 above)
Comment (mzajkowski) [reply src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs:48]: The longer I look at this approach the more I like it. It's really simple and based on the convention. The only concern would be with some more complex structures / themes with preprocessors etc. But still - they need to be compiled to the final stylesheet. See #7 to jump into another discussion related with the topic IMO.

I think we can leave it for now as it is and don't create problems before they really occur :)

File: src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs (hunk 1 above)
Comment (leekelleher) [reply src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs]: Fixed the property name casing.

File: src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs (hunk 2 above)
Comment (leekelleher) [reply src/Our.Umbraco.BackofficeThemes/Models/ThemeInfo.cs:48]: I've left this in for now - as I've said above, I'm happy for any alternative approaches for adding meta-data to a theme.

**Review by mzajkowski — APPROVED** [review]

**Pull request merged**